src/stringobj.c

```diff
@@ -1243,7 +1243,8 @@ Obj CopyToStringRep(
     copy = NEW_STRING(lenString);
 
     if ( IS_STRING_REP(string) ) {
-        memcpy(ADDR_OBJ(copy), CONST_ADDR_OBJ(string), SIZE_OBJ(string));
+        memcpy(CHARS_STRING(copy), CONST_CHARS_STRING(string),
+            GET_LEN_STRING(string));
```

Comment:
Good catch! This goes back to a255816 from Nov 2014, so it affects 4.10. On the up side, I imagine this problem cannot cause issues with GASMAN, only with Julia GC or Boehm, right? So it's not so bad (and less surprising that we never noticed).
Still, perhaps it should be backported to 4.10.

Reply:
With GASMAN, the overflow should indeed generally only hit unused memory, except possibly in the case of 32-bit systems, where it can possibly overflow the reserved area.

I did track it down on HPCGAP with the Boehm GC, where the change to `ImmutableString()` triggered it reliably during testing.

This may also be one candidate for the spurious crashes that we have seen with the Julia GC.